llama_index/node_parser/postprocessors/metadata_extractors.py

```diff
@@ -0,0 +1,398 @@
+                Prompt(
+                    template=f"""\
+{{context_str}}. Give {self._keywords} unique keywords for this \
+document. Format as comma separated. Keywords: """
+                ),
+                context_str=cast(TextNode, node).text,
+            )
+            node.metadata["excerpt_keywords"] = keywords
+
+
+class QuestionsAnsweredExtractor(MetadataExtractorBase):
+    """
+    Questions answered extractor. Node-level extractor.
+    Extracts `questions_this_excerpt_can_answer` metadata field.
+    Args:
+        llm_predictor (BaseLLMPredictor): LLM predictor
+        questions (int): number of questions to extract
+        prompt_template (str): template for question extraction
+    """
+
+    def __init__(
+        self,
+        llm_predictor: BaseLLMPredictor,
+        questions: int = 5,
+        prompt_template: Optional[str] = None,
+    ) -> None:
+        """Init params."""
+        if questions < 1:
+            raise ValueError("questions must be >= 1")
+        self._llm_predictor = llm_predictor
+        self._questions = questions
+        self._prompt_template = prompt_template
+
+    def __call__(self, nodes: Sequence[BaseNode]) -> None:
+        for node in nodes:
+            if self.is_text_node_only and not isinstance(node, TextNode):
+                continue
+            # Extract the title from the first node
+            questions = self._llm_predictor.predict(
+                Prompt(
+                    template=self._prompt_template
+                    or f"""\
+{{context_str}}. Given the contextual information, \
+generate {self._questions} questions this document can provide \
+specific answers to which are unlikely to be found elsewhere: \
+                """
+                ),
+                context_str=f"""\
+metadata: {json.dumps(node.metadata)} \
+content: {cast(TextNode, node).text}""",
+            )
+            node.metadata["questions_this_excerpt_can_answer"] = questions
+            # Only use this for the embedding
+            node.excluded_llm_metadata_keys = ["questions_this_excerpt_can_answer"]
+
+
+class SummaryExtractor(MetadataExtractorBase):
+    """
+    Summary extractor. Node-level extractor with adjacent sharing.
+    Extracts `section_summary`, `prev_section_summary`, `next_section_summary`
+    metadata fields
+    Args:
+        llm_predictor (BaseLLMPredictor): LLM predictor
+        self_summary (bool): whether to include self summary
+        prev_summary (bool): whether to include summary of previous node
+        next_summary (bool): whether to include summary of next node
+        embedding_only (bool): whether to only include summary in embedding
+            or show it to the user and to downstream predictors
+        prompt_template (str): template for summary extraction"""
+
+    def __init__(
+        self,
+        llm_predictor: BaseLLMPredictor,
+        self_summary: bool = True,
+        prev_summary: bool = False,
```

Comment:
The bool is not a very nice UX. Instead use, strings i.e. `summaries = ["prev", "current", "next"]`

Reply:
Can use the bools internally